acir/src/circuit/opcodes/black_box_function_call.rs

```diff
@@ -366,7 +373,11 @@ impl BlackBoxFuncCall {
                     })
                 }
             }
-            BlackBoxFunc::Pedersen => Ok(BlackBoxFuncCall::Pedersen { inputs, outputs }),
+            BlackBoxFunc::Pedersen => Ok(BlackBoxFuncCall::Pedersen {
+                inputs,
+                domain_separator: read_u32(&mut reader)?,
```

Comment:
I find the mixing of read with contructing the object messy.
This would be better with a common read input-outputs function and dedicated read/write functions per blackbox which can call the read input/outputs if needed.

Reply:
But since they are variants of the same enum the typing on the variant specifc functions would be pretty bad:
 ```rust
 fn read_aes<R: Read>(mut reader: R) -> std::io::Result<BlackBoxFuncCall> {
    let inputs = read_inputs(&mut reader)?;
    let outputs = read_outputs(&mut reader)?;
    Ok(BlackBoxFuncCall::AES { inputs, outputs })
}
 
 fn write_aes<W: Write>(aes: &BlackBoxFuncCall, mut writer: W) -> std::io::Result<()> {
    if let BlackBoxFuncCall::AES { inputs, outputs } = aes {
        write_inputs(inputs, &mut writer)?;
        write_outputs(outputs, &mut writer)?;
        Ok(())
    } else {
        Err(std::io::ErrorKind::InvalidData.into())
    }
}

 ```

Reply:
I think it works for the read(), but the write is not nice indeed.
What do you think of having the function for the read and keep the write as you did?